### PR TITLE
fabrics: ensure zero kato for non-persistent controllers

### PR DESCRIPTION
**Pull request by igaw**
We are missing a67d001330ae ("fabrics: ensure zero kato for non-persistent controllers") from the monolithic branch.

### CODE REVIEW

**Review by hreinecke — APPROVED** [review]

File: fabrics.c [hunk 1]
```diff
@@ -115,6 +115,16 @@ static void space_strip_len(int max, char *str)
 	}
 }
 
+static void set_discovery_kato(struct nvme_fabrics_config *cfg)
+{
+	/* Set kato to NVMF_DEF_DISC_TMO for persistent controllers */
+	if (persistent && !cfg->keep_alive_tmo)
```
Comment (hreinecke) [inline fabrics.c:121]: Where do you get the 'persistent' value from? Shouldn't it be rather an argument to the function?

Comment (igaw) [reply fabrics.c:121]: The persistent variable is a global bool. It's a bit ugly I agree. struct nvme_fabrics_config doesn't have it. We could add it there (libnvme) and also move this set_discovery_kato could move then there. Though I don't know if this makes sense.

**Comment by igaw** [comment]
I've decide to squash the two patches.

File: fabrics.c (hunk 1 above)
Comment (igaw) [reply fabrics.c:121]: I've added the patch in the meantime as it fixes a regression. We can still cleanup ontop of this change

**Pull request merged**